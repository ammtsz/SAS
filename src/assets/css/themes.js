import {
  colorDark,
  colorDarkBlue,
  colorWhite,
  colorLightGrey,
  colorBlack,
  colorLight,
  colorGrey,
  colorDarkGrey,
  colorDarkBlue70,
  colorDarkBlue50,
  colorDarkBlue40,
  colorDarkBlue20,
  colorLightBlue,
  colorLight08,
  colorDarkBlue08,
  colorBlack50,
} from "./variables";

export const LightTheme = {
  body: colorLightGrey,

  navbar: colorDarkBlue,

  cards: colorWhite,
  cardHover: colorGrey,

  shadow: colorDarkBlue40,

  border: colorLight,
  border2: colorGrey,

  text1: colorBlack,
  text2: colorDark,
  text3: colorDarkBlue,
  text4: colorDarkBlue70,
  text5: colorDarkGrey,

  icon: "",

  bg08: colorDarkBlue08,
  bg50: colorDarkBlue50,

  btn1: colorDarkBlue,
  btn2: colorDarkBlue50,
  btnHover: colorDarkBlue20,
  btnSwitcher: colorDarkBlue,
  btnTheme: colorWhite,

  progress1: colorDarkBlue20,
  progress2: colorDarkBlue,
};

export const DarkTheme = {
  body: colorBlack,

  navbar: colorDarkBlue,

  cards: colorDark,
  cardHover: colorBlack,

  shadow: colorLight,

  border: colorDark,
  border2: colorGrey,

  text1: colorWhite,
  text2: colorLight,
  text3: colorWhite,
  text4: colorGrey,
  text5: colorGrey,

  icon: "invert(60%) brightness(5)",

  bg08: colorLight08,
  bg50: colorBlack50,

  btn1: colorWhite,
  btn2: colorGrey,
  btnHover: colorDarkBlue,
  btnSwitcher: colorLightBlue,
  btnTheme: colorDarkGrey,

  progress1: colorGrey,
  progress2: colorLightBlue,
};
