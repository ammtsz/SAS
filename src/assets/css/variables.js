export const colorWhite = "#ffffff";
export const colorLight = "#e4e4e6";
export const colorLight08 = "rgba(255, 255, 255, 0.08)";
export const colorLightGrey = "#E5E5E5";
export const colorGrey = "#c9cccf";
export const colorDarkGrey = "#53595F";

export const colorBlack = "#000000";
export const colorBlack25 = "rgba(0, 0, 0, 0.25)";
export const colorBlack50 = "rgba(70, 70, 70, 0.5)";
export const colorDark = "#1E2124";
export const colorDarkBlue = "#343C58";
export const colorDarkBlue08 = "rgba(52, 60, 88, 0.08)";
export const colorDarkBlue20 = "rgba(52, 60, 88, 0.2)";
export const colorDarkBlue40 = "rgba(52, 60, 88, 0.4)";
export const colorDarkBlue50 = "rgba(52, 60, 88, 0.5)";
export const colorDarkBlue70 = "rgba(52, 60, 88, 0.7)";

export const colorBlue = "#0467db";
export const colorLightBlue = "#438DE4";
export const colorRed = "#FF4F4F";
export const colorGreen = "#32CB82";

export const mq_lg = "992px";
export const mq_md = "768px";
export const mq_sm = "576px";
export const mq_xs = "375px";

export const navbarHeight = "55px";
export const fontAvenir = "'Poppins', sans-serif";

export const cardMargin_mq_xs = `
@media (max-width: ${mq_xs}) {
    margin-left: 24px;
    margin-right: 24px;
  }
`;
