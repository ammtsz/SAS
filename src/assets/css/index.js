import styled from "styled-components";
import { colorWhite, colorBlue, colorDarkBlue } from "./variables.js";

export const CardContainer = styled.div`
  background-color: ${({ theme }) => theme.cards};
  border: 1px solid ${({ theme }) => theme.border};
  border-radius: 8px;
  box-sizing: border-box;
  box-shadow: 0px 1px 4px ${({ theme }) => theme.shadow};
`;

export const Button = styled.button`
  padding: 11px 24px;

  background: ${colorBlue};
  border: none;
  border-radius: 8px;

  color: ${colorWhite};
  font-size: 1rem;
  line-height: 1.5rem;

  cursor: pointer;
  outline: none;

  &:hover,
  &:focus {
    background: ${colorDarkBlue};
  }

  &[data-readonly="true"] {
    cursor: default;
    &:hover {
      background: ${colorBlue};
    }
  }
`;
