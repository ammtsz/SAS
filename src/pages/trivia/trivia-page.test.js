import React from "react";
import Trivia from "./trivia-page.component";

import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

import { MemoryRouter } from "react-router-dom";

import { colorDarkBlue } from "../../assets/css/variables";

describe("Page component <Trivia />", () => {
  const mockStore = configureMockStore();
  const mockQuestionsAnswered = [];
  const mockCurrentQuestion = {};
  const mockCurrentOptions = [];

  describe("not loading", () => {
    const store = mockStore({
      quiz: {
        category: { id: 1, name: "Vehicles", completed: 2 },
        questionsAnswered: mockQuestionsAnswered,
        currentQuestion: mockCurrentQuestion,
        currentOptions: mockCurrentOptions,
        loading: false,
        difficulty: "medium",
        questionNumber: 3,
      },
    });
    store.dispatch = jest.fn();

    let component;

    beforeEach(() => {
      component = render(
        <Provider store={store}>
          <MemoryRouter>
            <Trivia />
          </MemoryRouter>
        </Provider>
      );
    });

    afterEach(() => {
      cleanup();
    });

    it("should render Trivia", () => {
      expect(component.asFragment()).toMatchSnapshot();
    });

    it("should contain 'Check Answer' button", () => {
      expect(screen.getByText("Check Answer")).toBeInTheDocument();
    });

    it("should have 'Check Answer' button read-only", () => {
      const btn = screen.getByText("Check Answer");
      
      fireEvent.click(btn);

      expect(store.dispatch).toHaveBeenCalledTimes(0);
    });

    it("should NOT have 'Check Answer' button changing on mouseEnter", () => {
      const btn = screen.getByText("Check Answer");

      expect(btn).toHaveStyle({backgroundColor: ({ theme }) => theme.btn1});
      fireEvent.mouseEnter(btn);
      expect(btn).toHaveStyle({backgroundColor: ({ theme }) => theme.btn1});
    });

  });
});
