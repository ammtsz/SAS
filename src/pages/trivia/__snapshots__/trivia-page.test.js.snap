// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page component <Trivia /> not loading should render Trivia 1`] = `
<DocumentFragment>
  <section
    class="sc-iBPRYJ cpqzIC"
    data-testid="trivia-page"
  >
    <div
      class="sc-fubCfw gTGHKX"
    >
      <section
        class="sc-dlfnbm kMosoO"
        data-testid="modal-card-container"
        id="modal"
      >
        <div
          class="sc-bdfBwQ sc-hKgILt cXyUsK eptuml"
          data-testid="modal-card"
        >
          <img
            alt="incorrect"
            class="sc-eCssSg iCTmzf"
            src="wrong.svg"
          />
          <p
            class="sc-jSgupP ietSEv"
          >
            Wrong answer
          </p>
          <button
            class="sc-gsTCUz sc-gKsewC bijOHs jmdVDD"
            id="modal-button"
          >
            Next question
            <img
              alt="..."
              class="modal__nextButton--icon"
              src="arrow-right.svg"
            />
          </button>
        </div>
      </section>
      <div
        class="sc-iqHYGH cfIoLg"
      >
        <h4
          class="sc-crrsfI femVhl"
        >
          Vehicles
        </h4>
        <button
          class="sc-dQppl gXQlrE"
          data-testid="trivia-close-button"
        >
          <img
            alt="close"
            class="sc-bqyKva jGSKAU"
            src="close-btn.svg"
            tabindex="-1"
          />
          Close
        </button>
      </div>
      <article
        class="sc-bdfBwQ sc-pFZIQ cXyUsK egFQvc"
      >
        <div
          class="sc-kstrdz fCqIKu"
        >
          <h6
            class="sc-hBEYos fXGRYT"
            tabindex="0"
          >
            Question 3
          </h6>
          <div
            class="sc-fodVxV cNXSJY"
          >
            <span
              class="sc-fFubgz dpOLLe"
              tabindex="-1"
            >
              <img
                alt="star-level-easy"
                class=""
                src="star.svg"
              />
              <img
                alt="star-level-medium"
                class=""
                src="star.svg"
              />
              <img
                alt="star-level-hard"
                class="starless"
                src="star.svg"
              />
            </span>
            <span
              tabindex="0"
            >
              Medium
            </span>
          </div>
        </div>
        <h4
          class="sc-bkzZxe leMFja"
          tabindex="0"
        >
          undefined
        </h4>
        <div
          class="sc-jrAGrp jUyhyp"
          id="trivia-btn-answer-container"
        >
          <button
            class="sc-gsTCUz sc-kEjbxe bijOHs dlSfnO"
            disabled=""
            id="trivia-btn-answer"
          >
            Check Answer
          </button>
        </div>
      </article>
    </div>
  </section>
</DocumentFragment>
`;
